exporter/exporter.go

```diff
@@ -145,49 +144,28 @@ func main() {
 		log.Fatalf("failed to create database: %v", err)
 	}
 
-	consumers := make([]*Consumer, 0)
-	go func() {
-		tries := 0
-		nextTime := (time.Duration(math.Exp2(float64(tries))) * time.Millisecond) + time.Duration(rand.Intn(100))
-		timer := time.NewTimer(nextTime)
-		topics := config.Topics
-
-		for len(topics) > 0 {
-			var tmpTopics []TopicConfig
-
-			for _, topicConfig := range topics {
-				consumer, err := startConsumer(ctx, config.kafkaBrokers(), tlsConfig, httpClient, topicConfig)
-				if err != nil {
-					log.Printf("failed to start consumer with topic: %s: %v", topicConfig.Topic, err)
-					tmpTopics = append(tmpTopics, topicConfig)
-				} else {
-					consumers = append(consumers, consumer)
-				}
-			}
-			topics = tmpTopics
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt)
 
-			tries++
-			nextTime = (time.Duration(math.Exp2(float64(tries))) * time.Millisecond) + time.Duration(rand.Intn(100))
-			if nextTime > maxBackoff {
-				log.Printf("next timer %+v surpasses the max backoff time of %+v, setting to max backoff time\n", nextTime.String(), maxBackoff.String())
-				nextTime = maxBackoff
-			}
+	consumers := make([]*Consumer, 0)
+	for _, topicConfig := range config.Topics {
+		consumer, err := startConsumer(ctx, config.kafkaBrokers(), tlsConfig, httpClient, topicConfig)
+		if err != nil {
+			exitTime := (5 * time.Second) + time.Duration(rand.Intn(100))
```

Comment:
I don't see much value in jittering the exit time by 100ns.

Reply:
Fixed.